### ISSUE TITLE
Test flake: h2_sockpair_1byte_test@max_message_length@poller=poll-cv

### ISSUE THREAD

**Issue by guantaol** [grpc/grpc#17680]
https://source.cloud.google.com/results/invocations/1941929c-f199-448b-97d4-c93dd6ba2d03/targets/%2F%2Ftest%2Fcore%2Fend2end:h2_sockpair_1byte_test@max_message_length@poller%3Dpoll-cv/log

In `Bazel RBE Opt C/C++`:

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //test/core/end2end:h2_sockpair_1byte_test@max_message_length@poller=poll-cv
-----------------------------------------------------------------------------
D0109 00:53:35.920953718      19 test_config.cc:391]         test slowdown factor: sanitizer=1, fixture=2, poller=5, total=10
D0109 00:53:35.921310277      19 ev_posix.cc:170]            Using polling engine: poll-cv
D0109 00:53:35.921400845      19 dns_resolver.cc:334]        Using native dns resolver
I0109 00:53:35.921545854      19 max_message_length.cc:103]  testing request with send_limit=0 use_service_config=0 use_string_json_value=0
I0109 00:53:35.921563196      19 max_message_length.cc:44]   Running test: test_max_request_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.922322622      19 max_message_length.cc:103]  testing request with send_limit=1 use_service_config=0 use_string_json_value=0
I0109 00:53:35.922367554      19 max_message_length.cc:44]   Running test: test_max_request_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.922543735      19 max_message_length.cc:294]  testing response with send_limit=0 use_service_config=0 use_string_json_value=0
I0109 00:53:35.922559584      19 max_message_length.cc:44]   Running test: test_max_response_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.922972325      19 max_message_length.cc:294]  testing response with send_limit=1 use_service_config=0 use_string_json_value=0
I0109 00:53:35.922989493      19 max_message_length.cc:44]   Running test: test_max_response_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.923362208      19 max_message_length.cc:103]  testing request with send_limit=1 use_service_config=1 use_string_json_value=0
I0109 00:53:35.923379505      19 max_message_length.cc:44]   Running test: test_max_request_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.923697018      19 max_message_length.cc:103]  testing request with send_limit=1 use_service_config=1 use_string_json_value=1
I0109 00:53:35.923713118      19 max_message_length.cc:44]   Running test: test_max_request_message_length/chttp2/socketpair_one_byte_at_a_time
I0109 00:53:35.923978480      19 max_message_length.cc:294]  testing response with send_limit=0 use_service_config=1 use_string_json_value=0
I0109 00:53:35.923994346      19 max_message_length.cc:44]   Running test: test_max_response_message_length/chttp2/socketpair_one_byte_at_a_time
E0109 00:53:35.924255771      19 cq_verifier.cc:273]         cq returned unexpected event: OP_COMPLETE: tag:0x1 OK
E0109 00:53:35.924306921      19 cq_verifier.cc:280]         expected tags:
tag(101) GRPC_OP_COMPLETE success=1 test/core/end2end/tests/max_message_length.cc:418
*******************************
Caught signal SIGABRT
test/core/end2end/h2_sockpair_1byte_test[0x438c75]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7f1f682a7390]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x38)[0x7f1f67660428]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x16a)[0x7f1f6766202a]
test/core/end2end/h2_sockpair_1byte_test[0x437fab]
test/core/end2end/h2_sockpair_1byte_test[0x42272f]
test/core/end2end/h2_sockpair_1byte_test[0x421a42]
test/core/end2end/h2_sockpair_1byte_test[0x40576a]
test/core/end2end/h2_sockpair_1byte_test[0x40359c]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f1f6764b830]
test/core/end2end/h2_sockpair_1byte_test[0x403459]
Aborted (core dumped)
```

**Comment by yashykt** [comment]
The issue is described and fixed in the pending PR #18272

**Comment by jtattermusch** [comment]
poll-cv was removed in #18298
